libs/community/langchain_community/vectorstores/pgvector.py

```diff
@@ -505,22 +506,64 @@ def __query_collection(
 
             if filter is not None:
                 filter_clauses = []
-                IN, NIN = "in", "nin"
+                IN, NIN, BETWEEN, GT, LT, NE, EQ, LIKE, CONTAINS, OR, AND = "in", "nin", "between", "gt", "lt", "ne", "eq", "like", "contains", "or", "and"
+
+                def create_clause(value):
```

Comment:
could this be its own method on the class? would make it more readable for sure

Reply:
Yes of course :)